pxr/usd/lib/usdGeom/points.cpp

```diff
@@ -238,4 +238,82 @@ TF_REGISTRY_FUNCTION(UsdGeomBoundable)
         _ComputeExtentForPoints);
 }
 
+size_t
+UsdGeomPoints::ComputePositionsAtTimes(
+    std::vector<VtVec3fArray>& positions,
+    const std::vector<UsdTimeCode>& sampleTimes,
+    UsdTimeCode baseTime,
+    float velocityScale) const {
+    constexpr double epsilonTest = 1e-5;
+    const auto sampleCount = sampleTimes.size();
+    if (sampleCount == 0 || positions.size() < sampleCount || baseTime.IsDefault()) {
+        return 0;
+    }
+
+    const auto pointsAttr = GetPointsAttr();
+    if (!pointsAttr.HasValue()) {
+        return 0;
+    }
+
+    bool hasValue = false;
+    double pointsLowerTimeSample = 0.0;
+    double upperTimeSample = 0.0;
+    if (!pointsAttr.GetBracketingTimeSamples(baseTime.GetValue(), &pointsLowerTimeSample,
+                                             &upperTimeSample, &hasValue) || !hasValue) {
+        return 0;
+    }
+
+    VtVec3fArray points;
+    VtVec3fArray velocities;
+
+    // We need to check if there is a queriable velocity at the given point,
+    // if it has almost the same lower time sample and the array lengths are equal.
```

Comment:
Right, I forgot about the uses of a constant velocity field, or varying frequency. Updated the code to require both times to match.